scripts/install_deps.sh

```diff
@@ -0,0 +1,116 @@
+apt_deps=$(cat ${ATOS_REPO_PATH}/scripts/dependencies.txt | tr '\n' ' ')
+echo "Installing dependencies... $apt_deps"
+sudo apt update && sudo apt install -y ${apt_deps}
+
+# Check if apt failed to install dependencies
+check_command_failed $? "Failed to install dependencies."
+
+#######################################
+###### Install ROS2 dependencies ######
+#######################################
+
+# Check if the ROS2 repository is already added
+if ! (dpkg -l | grep -q "ros-$ROS_DISTRO-desktop"); then
+    echo "Adding the ROS2 $ROS_DISTRO apt repository..."
+
+    # Install ROS2 prerequisites
+    sudo apt update && sudo apt install -y lsb-release ros-dev-tools
+
+    # Authorize the ROS2 gpg key with apt
+    sudo curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key \
+    -o /usr/share/keyrings/ros-archive-keyring.gpg
+
+    # Add the ROS2 repo to sources list
+    echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(source /etc/os-release && echo $UBUNTU_CODENAME) main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
+else
+    echo "ROS2 $ROS_DISTRO" repository already added, skipping addition...
+fi
+
+# Install ROS2 packages
+echo "Installing ROS2 packages..."
+sudo apt install -y \
+    ros-${ROS_DISTRO}-desktop \
+    ros-${ROS_DISTRO}-geographic-msgs \
+    ros-${ROS_DISTRO}-geometry-msgs \
+    ros-${ROS_DISTRO}-std-msgs \
+    ros-${ROS_DISTRO}-std-srvs \
+    ros-${ROS_DISTRO}-nav-msgs \
+    ros-${ROS_DISTRO}-geographic-msgs \
+    ros-${ROS_DISTRO}-foxglove-msgs \
+    ros-${ROS_DISTRO}-pcl-conversions \
+    ros-${ROS_DISTRO}-rosbridge-suite
+check_command_failed $? "Failed to install ROS2 packages."
+
+if [ "$ROS_DISTRO" == "humble" ]; then
+    sudo apt install -y libpaho-mqtt-dev
+elif [ "$ROS_DISTRO" == "foxy" ]; then
+    sudo apt install -y ros-foxy-paho-mqtt-c
+fi
+
+###########################################
+###### Install some deps from source ######
+###########################################
+
+# Set custom path for source installation, first check if it exists
+SOURCE_PATH=$HOME/temp/atos_install
+if [ -d "$SOURCE_PATH" ]; then
+    echo "Removing preexisting path ${SOURCE_PATH} for source installation..."
+    sudo rm -rf $SOURCE_PATH
+fi
+
+echo "Creating custom path for source installation at $SOURCE_PATH"
+mkdir -p $SOURCE_PATH
+
+# Install OpenSimulationInterface but first check if the library is already installed
+if [ -d "/usr/local/lib/osi3" ]; then
+    echo "OpenSimulationInterface already installed, skipping installation..."
+else
+    echo "Installing OpenSimulationInterface..."
+    git clone --depth 1 --branch v3.4.0 https://github.com/OpenSimulationInterface/open-simulation-interface.git $SOURCE_PATH/open-simulation-interface
+    cd $SOURCE_PATH/open-simulation-interface
+    mkdir -p build && cd build
+    cmake .. && make -j8
```

Comment:
can hardcoded j8 cause problems here? maybe use nproc

Reply:
Causes issues in the Docker build otherwise. Having high j value then available processors will not cause an issue.

Reply:
Maybe github actions builds could be sped up by setting lower jX values. Could be investigated!